internal/attestation/renderer/renderer.go

```diff
@@ -134,6 +152,48 @@ func (ab *AttestationRenderer) Render() (*dsse.Envelope, error) {
 	return &dsseEnvelope, nil
 }
 
+// addPolicyResults adds policy evaluation results to the statement. It does it by deserializing the predicate from a structpb.Struct,
+// filling PolicyEvaluations, and serializing it again to a structpb.Struct object, using JSON as an intermediate representation.
+// Note that this is needed because intoto predicates are generic structpb.Struct
+func addPolicyResults(statement *intoto.Statement, policyResults []*v1.PolicyEvaluation) error {
+	predicate := statement.Predicate
+	// marshall to json
+	jsonPredicate, err := protojson.Marshal(predicate)
+	if err != nil {
+		return fmt.Errorf("marshalling predicate: %w", err)
+	}
+
+	// unmarshall to our typed predicate object
+	var p chainloop.ProvenancePredicateV02
+	err = json.Unmarshal(jsonPredicate, &p)
+	if err != nil {
+		return fmt.Errorf("unmarshalling predicate: %w", err)
+	}
+
+	// insert policy evaluations
+	if p.PolicyEvaluations == nil {
+		p.PolicyEvaluations = make(map[string][]*v1.PolicyEvaluation)
+	}
+	p.PolicyEvaluations["ATTESTATION"] = policyResults
```

Comment:
You should use the string representation of the MaterialType enum to avoid future inconsistencies

```suggestion
	p.PolicyEvaluations[schemaapi.CraftingSchema_Material_ATTESTATION.String()] = policyResults
```

Reply:
Ey, thanks for the suggestion @jotadrilo . But in this case it's not a material, but policies applied to the whole attestation. I know the naming is confusing, we might choose a different key for this particular case. I'll give it a try.

Reply:
But you are mapping materials and policies, right? In other groups, you are grouping policies by their material name.

If this specific grouping has a different semantic, I'd better use a different field in the in-toto predicate (v2), or add a reserved MaterialType name for this kind of grouping.

Reply:
Yes, we were evaluating those options. We'll revisit this in next iterations, as it might be part of a batch with other improvements. Thanks!